Include/internal/pycore_code.h

```diff
@@ -416,33 +429,75 @@ extern PyObject* _Py_GetSpecializationStats(void);
 #ifdef WORDS_BIGENDIAN
 
 static inline void
-write32(uint16_t *p, uint32_t val)
+write_u32(uint16_t p[2], uint32_t val)
 {
     p[0] = val >> 16;
     p[1] = (uint16_t)val;
 }
 
 static inline uint32_t
-read32(uint16_t *p)
+read_u32(uint16_t p[2])
 {
     return (p[0] << 16) | p[1];
 }
 
+static inline void
+write_obj(uint16_t p[4], PyObject *obj)
```

Comment:
Could you add `write_u64` and `read_u64`, and then use that as follows:
```C
static inline void write_obj(...) {
    uintptr_t val = (uintptr_t)obj;
#if SIZEOF_VOID_P == 8
    write_u64(p, val);
#else
#if SIZEOF_VOID_P == 4
    write_u32(p, val);
#else
    ERROR...
```

Likewise for the read.